ultranest/popstepsampler.py

```diff
@@ -688,7 +688,333 @@ def __next__(
             return None, None, None, nc
 
 
+
+
+def slice_limit_to_unitcube(tleft, tright):
+    
+    """
+    return the slice limits as a copy of intersection between the slice and the unit cube boundaries
+
+
+    parameters
+    ----------
+        tleft: float
+                Intersection of the slice and the unit cube boundaries in the inverse direction of the slice
+        tright: float
+                Intersection of the slice and the unit cube boundaries in the direction of the slice
+    Returns
+    -------
+        (tleft_new,tright_new): tuple
+                Positive and negative slice limits
+    """
+    tleft_new,tright_new = tleft.copy(),tright.copy()
+    return (tleft_new,tright_new)
+
+
+def slice_limit_to_scale(tleft, tright):
+
+    """
+    return the slice limits as an interval of size `2*scale` or the intersection between the slice and the unit cube boundaries
+    if the interval is larger than the unit cube boundaries.
+
+    parameters
+    ----------
+        tleft: float
+                Intersection of the slice and the unit cube boundaries in the inverse direction of the slice
```

Comment:
Intersection of the unit cube with the slice in the negative direction

Intersection of the unit cube with the slice in the positive direction